include/psa/crypto_types.h

```diff
@@ -65,10 +65,24 @@ typedef int32_t psa_status_t;
  */
 typedef uint32_t psa_key_type_t;
 
-/** The type of PSA elliptic curve identifiers. */
+/** The type of PSA elliptic curve identifiers.
+ *
+ * The encoding of curve identifiers is aligned with the
+ * TLS Supported Groups Registry (formerly known as the
+ * TLS EC Named Curve Registry)
+ * https://www.iana.org/assignments/tls-parameters/tls-parameters.xhtml#tls-parameters-8
+ * The values are defined by RFC 8422 and RFC 7027.
```

Comment:
The values defined in this specification, but the IANA registry has more and may add more in the future.

I think we should take this one step further and specify the IANA registry is included by reference with respect to numerical values. This specification only contributes the names, which are generally `PSA_ECC_CURVE_` + uppercase(IANA.description), but there are exceptions (additional underscore for Brainpool, `CURVE_` rather than `x` for 25519 and 448).

Thus vendors who want to support other curves that are in the IANA registry should use the IANA value and a vendor-specific name. Vendors who want to support non-IANA curves should use the vendor range.

And we should define a vendor range that is a subset of the IANA private use range (which is only 0x01fc..0x01ff and 0xfe00..0xfeff).

Reply:
Is it worth combining the `psa_dh_group_t` and `psa_ecc_curve_t` types, given that they are both extracted from the same enumeration in the IANA registry? Then we can provide a single overall description of the association with the registry regarding the symbols/values defined in this spec, the use of other values by vendors and the right way to declare non-IANA curves.

It might be helpful to also move the type(s) and the ECC and DH definitions into a separate subsection of the API reference, so they clearly are connected to these overall description?

Reply:
I prefer to keep them separate because there's no context in which you can pass either. Not that C enforces integral types anyway, but they're semantically distinct.

Reply:
> I think we should take this one step further and specify the IANA registry is included by reference with respect to numerical values.

Some specific examples to clarify your suggestion. Is the proposal that all defined IANA registry values are "in scope" of the specification when combined with the appropriate ECC or DH keys types - and therefore not "vendor defined"? For example:

- The key type for a public key of an IANA registered ECC curve, but which is not explicitily provided an identifier in this specification, would have the value `0x6003xxxx`. It would not require the vendor flag to be set (`0xE003xxxx`).

Would the key type for a non-IANA ECC curve, require:
1. Use of a value in the IANA private use range for the curve id.
2. Use of the vendor flag in the key type.
3. Either (1) or (2).
4. Both (1) and (2). Although, presumably if the vendor flag is used, the use of a private use id and the `0x6003xxxx` key type is just a recommendation.

Reply:
(3), I think. Curve encodings have their own existence independently of key type encodings.

Reply:
Just wrote that all up in some follow-up commits.

I've sliced off half of each of the private-use ranges for vendor-defined identifiers, leaving half for future potential use by the PSA specification. Let me know if that approach does not make sense...

Reply:
I am not an expert in this material, and happily defer to your judgement. Are you suggesting that I just remove the line with the RFC reference:

> * The values are defined by RFC 8422 and RFC 7027.

Reply:
Is the same comment applicable to the RFC 7919 reference in the documenation for `psa_dh_group_t`?

Reply:
Yes, please remove this line and the similar line for `psa_dh_group_t`.